src/js/background/tab_state_tracker/tabStateTracker.ts

```diff
@@ -47,3 +47,15 @@ export function updateContentScriptState(
     newState,
   );
 }
+
+export function invalidateContentScriptStateAndThrow(
+  sender: ValidSender,
+  origin: Origin,
+  error: string,
+): never {
+  getOrCreateTabStateMachine(sender.tab.id, origin).updateStateForFrame(
```

Comment:
Can we leverage the method above?

Reply:
The helper I built assumes a Map of Maps structure, where this is a Map of StateMachine. There's probably a way to make this work with interfaces and maybe an extra callback for what the default value should be, but I'm not sure the value is there.